src/test/make/docker-helper.sh

```diff
@@ -0,0 +1,171 @@
+  GIT_ROOT=$(git rev-parse --show-toplevel)
+fi
+: ${GIT_ROOT}
+DIR=$(realpath --relative-to "$GIT_ROOT" .)
+
+: ${DOCKER:=podman}
+: ${DOCKER_PLATFORM:=--platform linux\/amd64}
+: ${DOCKER_USER:=root}
+: ${DOCKER_IMAGE:=ghcr.io/uq-pac/basil-tools-docker}
+
+if [[ $# -lt 1 ]] || [[ "$1" == --help ]]; then
+  echo "usage: $(basename $0) (pull | push | build | start | stop | shell | hash | env [--unset] | COMMAND...)"
+  ! [[ $# -lt 1 ]]
+  exit
+fi
+
+DOCKER_CMD="$(realpath $0)"
+
+if [[ -z "${DOCKER_FLAKE:-}" ]] && [[ -r "$(dirname $DOCKER_CMD)/docker-flake.txt" ]]; then
+  DOCKER_FLAKE="$(cat $(dirname $DOCKER_CMD)/docker-flake.txt)"
+fi
+
+: $DOCKER_FLAKE
+
+# create unique names depending on the flake reference, to ensure the correct container
+# is used.
+# unique names depend only on DOCKER_FLAKE, allowing them to be computed without nix.
+commit=$(printf '%s' "$DOCKER_FLAKE" | grep --only-matching -E '[0-9a-fA-F]{40}' | head -c8)
+flake_hash=flake-$(printf '%s' "$DOCKER_FLAKE" | md5sum | cut -d' ' -f1 | head -c4)
+
+if [[ -z "${DOCKER_TAG:-}" ]]; then
+  DOCKER_TAG="$flake_hash-$commit"
+fi
+
+unique_image="$DOCKER_IMAGE:$DOCKER_TAG"
+unique_container="container-$DOCKER_TAG"
+
+# this allows the env subcommand to output syntax compatible with multiple shells
+shell=$(basename $SHELL)
+if [[ $shell == fish ]]; then
+  unset='set --erase'
+  unalias='functions --erase'
+  eval='('
+else
+  unset=unset
+  unalias=unalias
+  eval='$('
+fi
+
+
+if [[ "$1" == pull ]]; then
+  # pulls the unique image from the registry
+  set -x
+  exec $DOCKER pull $DOCKER_PLATFORM "$unique_image"
+
+elif [[ "$1" == push ]]; then
+  # pushes the unique image to the registry. image must already exist locally.
+  set -x
+  exec $DOCKER push "$unique_image"
+
+elif [[ "$1" == build ]]; then
+  # builds the docker image for running tools.
+  # safe to re-run. if docker image is already up-to-date, should be reasonably fast.
+  nix build "$DOCKER_FLAKE" --no-link
+  nix build "$DOCKER_FLAKE.conf" --no-link
+  conf=$(nix build "$DOCKER_FLAKE.conf" --no-link --print-out-paths)
+  tag=$(nix eval --expr "with builtins; (fromJSON (unsafeDiscardStringContext (readFile $conf))).repo_tag" --impure --raw)
+  if ! [[ "$tag" == "$DOCKER_IMAGE":* ]]; then
+    printf '%s %s %s.\n' \
+      "ERROR: docker image names do not match!" \
+      "nix flake will build '$tag', but" \
+      "DOCKER_IMAGE is '$DOCKER_IMAGE'" >&2
+    exit 1
+  fi
+  set -x
+  $(nix build "$DOCKER_FLAKE" --no-link --print-out-paths) | "$DOCKER" image load
+  $DOCKER image tag "$tag" $unique_image
+  exit
+
+elif [[ "$1" == start ]]; then
+  # starts an instance of the docker image.
+  set -x
+  exec $DOCKER run $DOCKER_PLATFORM -v"$GIT_ROOT:$GIT_ROOT" --rm -td --user $DOCKER_USER --name $unique_container $unique_image
+
+elif [[ "$1" == stop ]]; then
+  # starts the instance of the docker image.
+  set -x
+  exec $DOCKER rm -f -t 1 $unique_container
```

Comment:
This doesn't work with docker, only podman, since docker doesn't have a -t flag for rm